### PR TITLE
Add Parallels child detection

### PR DESCRIPTION
**Pull request by JSCU-CNI**
This PR adds Parallels child detection for MacOS targets.

### CODE REVIEW

File: dissect/target/plugins/child/parallels.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    "Library/Group Containers/*.com.parallels.desktop.appstore/Shared/Parallels",
+]
+
+PARALLELS_SYSTEM_PATHS = [
+    "/Users/Shared/Parallels",
+]
+
+
+def find_pvms(target: Target) -> Iterator[TargetPath]:
+    """Finds virtual machines located in default folders on a MacOS target.
```
Comment (Schamper) [inline dissect/target/plugins/child/parallels.py:21]:
```suggestion
    """Finds virtual machines located in default folders on a macOS target.
```

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/plugins/child/parallels.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            if "*" in parallels_path:
+                start_path, pattern = parallels_path.split("*", 1)
+                for path in user_details.home_path.joinpath(start_path).rglob("*" + pattern):
+                    yield from iter_vms(path)
+            else:
+                if (path := user_details.home_path.joinpath(parallels_path)).exists():
+                    yield from iter_vms(path)
+
+
+def iter_vms(path: TargetPath) -> Iterator[TargetPath]:
```
Comment (Schamper) [inline dissect/target/plugins/child/parallels.py:41]:
```suggestion
def iter_vms(path: Path) -> Iterator[TargetPath]:
```

File: dissect/target/plugins/child/parallels.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.pvms = list(find_pvms(target))
+
+    def check_compatible(self) -> None:
+        if not self.pvms:
+            raise UnsupportedPluginError("No Parallels pvm file(s) found")
+
+    def list_children(self):
```
Comment (Schamper) [inline dissect/target/plugins/child/parallels.py:61]:
```suggestion
    def list_children(self) -> Iterator[ChildTargetRecord]:
```

File: dissect/target/plugins/child/parallels.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+from typing import Iterator
```
Comment (Schamper) [inline dissect/target/plugins/child/parallels.py:1]:
```suggestion
from pathlib import Path
from typing import Iterator
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Review by Schamper — CHANGES_REQUESTED** [review]
If you fix the linter in `tests/plugins/child/test_parallels.py` this is good to go.

**Comment by Schamper** [comment]
I'm going on vacation and unfortunately don't have time to look at this now, I'll see if someone from the team can help :smile:

**Comment by twiggler** [comment]
@JSCU-CNI I have created a fix for the root cause in a separate PR: https://github.com/fox-it/dissect.target/pull/867

After fix has been merged, you can rebase so that your unit test should pass when running on windows.

**Comment by twiggler** [comment]
@JSCU-CNI https://github.com/fox-it/dissect.target/pull/867 has been merged, you can rebase

**Review by Schamper — APPROVED** [review]

**Pull request merged**